crates/bevy_diagnostic/src/system_diagnostics.rs

```diff
@@ -0,0 +1,120 @@
+use std::any::TypeId;
+
+use super::{Diagnostic, DiagnosticId, Diagnostics};
+use bevy_app::prelude::*;
+use bevy_ecs::{
+    ArchetypeComponent, BoxedSystem, Resources, System, SystemId, ThreadLocalExecution, TypeAccess,
+    World,
+};
+use bevy_utils::Instant;
+
+pub trait AppBuilderMeasuredSystemExt {
+    /// Add a system and record its execution time in [Diagnostics].
+    ///
+    /// # Example
+    /// ```
+    /// use bevy::{diagnostic::{AppBuilderMeasuredSystemExt, LogDiagnosticsPlugin}, prelude::*};
+    ///
+    /// pub fn timed_system() {
+    ///     std::thread::sleep(std::time::Duration::new(0, 5000000));
+    /// }
+    ///
+    /// fn main() {
+    ///     App::build()
+    ///         .add_plugins(DefaultPlugins)
+    ///         .add_plugin(LogDiagnosticsPlugin::default())
+    ///         .add_measured_system("timed_system", timed_system.system())
```

Comment:
If possible, I would prefer a `.add_system(timed_system.measured_system("timed_system"))`
That way, all methods adding systems (`add_system`, `add_system_to_stage`, `add_startup_system`, ...) won't have to be redone in a `measured` variant

Reply:
Good suggestion, the new commit implements that with this syntax: `add_system(timed_system.system().measured())`